### PR TITLE
Add a README detailing all the available options

### PR DESCRIPTION
**Pull request by steveburton4**
Add a README detailing all the available options

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,118 @@
...
-Lambda function to send CloudWatch alerts to Slack
+Lambda function to send CloudWatch alerts to Slack written in python 3.8
+
+## Basic functionality
+
+The lambda reacts to given SNS messages, looks at the payload and generates a slack message payload which it then posts to a slack webhook URL.
+
+## Message structure
+
+The lambda will react to alerts sent from SNS to whatever topics it is subscribed to. The SNS message stucture can be found at https://docs.aws.amazon.com/lambda/latest/dg/with-sns.html.
```
Comment (tea-short) [inline README.md:10]: stRucture

File: README.md [hunk 2]
```diff
@@ -1,2 +1,118 @@
...
+
+## Different message types
+
+The app will look at the message payload (`$.Records[0].SNS.Message`) and react differently depending on different fields that are present. These scenarios are desribed below.
+
+If none of these scenarios are matched, then there is a fallback scenario which will which send a message with the phrase `Unindentified message` as the title the dumped payload as well.
```
Comment (tea-short) [inline README.md:16]: UniNdentified message

File: README.md [hunk 3]
```diff
@@ -1,2 +1,118 @@
...
+* `Payment`
+* `Pipeline`
+* `TransferToPensionAge`
+* `MI12CaseControl`
+* `QLR`
+* `ReleaseComparison`
+
+### Guard duty notifications
+
+If the message has the `detail-type` field and the field contains `GuardDuty Finding` it is a guard duty notification. If this is the case then a notification is logged with the text `AWS GuardDuty Finding Type` and a URL is consructed to go directly to the finding in the AWS console.
```
Comment (tea-short) [inline README.md:41]: consTructed

File: README.md [hunk 4]
```diff
@@ -1,2 +1,118 @@
...
+
+### Configuration notifications
+
+If the message has the `configRuleName` field it is a configuration notification. If this is the case then a notification is logged with the text `AWS Config Compliance Change detected` and a URL is consructed to go directly to the finding in the AWS console.
+
+### Cloudwatch event notifications
+
+If the message has the `messageType` field it is a cloudwatch event notification. If this is the case then a notification is logged with the text `AWS Config CloudWatch Event` and a URL is consructed to go directly to the finding in the AWS console. Details of the event are pulled out and provided as extra information in the slack message.
```
Comment (tea-short) [inline README.md:49]: consTructed

Comment (steveburton4) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (steveburton4) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (steveburton4) [reply README.md]: Fixed

File: README.md [hunk 5]
```diff
@@ -1,2 +1,118 @@
...
+
+### Cloudwatch alarm notifications
+
+If the message has the `AlarmName` field it is a cloudwatch alarm notification. If this is the case their are two sub scenarios based on the message itself.
+
+#### Prowler alarms
+
+If the message has the `Namespace` field and its value is set to `Prowler/Monitoring` then the message is assumed to be from the Prowler monitoring that is set up. In this case, the message is processed with a title of the name of the monitoring alert triggered an a URL to go to the alert directly is generated and set in the slack message.
```
Comment (tea-short) [inline README.md:57]: anD a URL

File: README.md [hunk 6]
```diff
@@ -1,2 +1,118 @@
...
+
+The icon is set from the severity and the type of notification. These values also set the channel that the slack notification goes to and they are logged in the message as well. Their values are retrieved from the tags on the cloudwatch alarm itself and are detailed in the section below.
+
+#### Custom alarm tags
+
+The following tags are retrieved from the cloudwatch alarm and each one has a default in case it is not present:
+
+* `severity` -> this is used to determine the icon used for the alarm and the channel it goes to, supported values are `Low`, `Medium`, `High` and `Critical`. Default is `Medium`
+
+* `notification_type` -> this is used to determine the icon used for the alarm and the channel it goes to, supported values are `Information`, `Warning` and `Error`. Default is `Warning`
```
Comment (tea-short) [inline README.md:75]: Descriptions for both `severity` and `notification_type` are the same but it's not clear how they are different / why both exist if they are the same

File: README.md (hunk 5 above)
Comment (steveburton4) [reply README.md]: Fixed

File: README.md (hunk 6 above)
Comment (steveburton4) [reply README.md]: I've added a bit saying it's a combination of both that is used. The combinations are details in the `Environment variables` section so have pointed it to there too.

**Review by tea-short — APPROVED** [review]

**Pull request merged**